kernel/src/device/char.rs

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! A subsystem for character devices (or char devices for short).
+
+#![expect(dead_code)]
+
+use device_id::{DeviceId, MajorId};
+use inherit_methods_macro::inherit_methods;
+
+use crate::{
+    events::IoEvents,
+    fs::{
+        device::{add_node, Device, DeviceType},
+        fs_resolver::FsResolver,
+        inode_handle::FileIo,
+        utils::{InodeIo, StatusFlags},
+    },
+    prelude::*,
+    process::signal::{PollHandle, Pollable},
+};
+
+/// A character device.
+pub trait CharDevice: Send + Sync + Debug {
+    /// Returns the name of the char device.
+    fn name(&self) -> &str;
```

Comment:
Return `DevtmpfsName` as suggested in #2562. 

[The Evdev PR](https://github.com/asterinas/asterinas/pull/2561) needs to add inodes whose paths are `/dev/input/eventX`. Simply returning a device name of `&str` does not provide the sufficient information for the full path like `/dev/input/eventX`.